lib/util/exec/index.ts

```diff
@@ -107,7 +107,7 @@ export async function exec(
     const dockerOptions = {
       ...docker,
       cwd,
-      envVars: dockerEnvVars(extraEnv, childEnv),
+      envVars: dockerEnvVars(extraEnv, childEnv).concat(docker.envVars),
```

Comment:
@rarkins I'm resuming my work on this one, I have my code rebased with master and most of the code review changes done.

I'm having a hard time understanding how `dockerEnvVars` should work. That function will basically extract the keys for the envVariables in `childEnv`. So, if I have some variables like this:

```
{
 "BUNDLE_GEMS__PRIVATE__COM": "some-user:some-password"
}
```

It will return an array with `["BUNDLE_GEMS__PRIVATE__COM"]`. 

Shouldn't this method generate an array like this one:

```
["BUNDLE_GEMS__PRIVATE__COM=some-user:some-password"]
```

If not the env variables are not getting passed correctly to the docker command builder. I might be missing something here or there might be another way to set them.

Reply:
@zharinov can you help out on this one?

Reply:
Example test case that starts failing with the change:

```
  ● .updateArtifacts() › supports docker mode with credentials

    expect(received).toMatchSnapshot()

    Snapshot name: `.updateArtifacts() supports docker mode with credentials 1`

    - Snapshot  - 2
    + Received  + 2

    @@ -4,18 +4,18 @@
          "options": Object {
            "encoding": "utf-8",
          },
        },
        Object {
    -     "cmd": "docker run --rm --user=foobar -v \"/tmp/github/some/repo\":\"/tmp/github/some/repo\" -v \"/tmp/renovate/cache\":\"/tmp/renovate/cache\" -v \"/tmp/renovate/cache/others/go\":\"/tmp/renovate/cache/others/go\" -e GOPATH -e GOPROXY -e GONOSUMDB -e CGO_ENABLED -w \"/tmp/github/some/repo\" renovate/go bash -l -c \"git config --global url.\\\"https://some-token@github.com/\\\".insteadOf \\\"https://github.com/\\\" && go get -d ./...\"",
    +     "cmd": "docker run --rm --user=foobar -v \"/tmp/github/some/repo\":\"/tmp/github/some/repo\" -v \"/tmp/renovate/cache\":\"/tmp/renovate/cache\" -v \"/Users/bilby91/go\":\"/Users/bilby91/go\" -e GOPATH=/Users/bilby91/go -e GOPROXY=proxy.example.com -e GONOSUMDB=1 -e CGO_ENABLED=1 -w \"/tmp/github/some/repo\" renovate/go bash -l -c \"git config --global url.\\\"https://some-token@github.com/\\\".insteadOf \\\"https://github.com/\\\" && go get -d ./...\"",
```

What would be the purpose of doing `-e GOPATH` instead of `-e GOPATH=/Users/bilby91/go` ?

Reply:
I think that your `GOPATH` may be getting passed through to the test case, so it ends up with a different snapshot. Need to either unset GOPATH somehow or avoid using a snapshot for that test.

Reply:
@rarkins I think I learned a new behaviour of `docker run` that I was not aware. Doing `-e MY_VAR` will pick the value from the current `env`. I thought that this was not supported and you always had to do `-e MY_VAR=my-value`. I'll get the changes done now!